base/reflection.jl

```diff
@@ -862,19 +862,29 @@ function MethodList(mt::Core.MethodTable)
 end
 
 """
-    methods(f, [types])
+    methods(f, [types], [module])
 
-Returns the method table for `f`.
+Return the method table for `f`.
 
-If `types` is specified, returns an array of methods whose types match.
+If `types` is specified, return an array of methods whose types match.
+If `module` is specified, return an array of methods defined in this module.
+A list of modules can also be specified as an array or tuple.
+
+!!! compat "Julia 1.4"
+    At least Julia 1.4 is required for specifying a module.
 """
-function methods(@nospecialize(f), @nospecialize(t))
+function methods(@nospecialize(f), @nospecialize(t),
+                 @nospecialize(mod::Union{Module,AbstractArray{Module},Tuple{Vararg{Module}}}=()))
```

Comment:
I realize that by passing `()` explicitly, like in `methods(f, (Int,), ())`, the user would expect an empty result, whereas here we consider it means no filtering on the module... should we change the default to `nothing`? This use-case is very contrived, but it might happen programmatically... Hm, I think I will use `nothing`, it's more correct.